debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/OracleConnection.java

```diff
@@ -134,7 +134,7 @@ private OracleDatabaseVersion resolveOracleDatabaseVersion() {
             }
             catch (SQLException e) {
                 // exception ignored
-                if (e.getMessage().contains("ORA-00904: \"BANNER_FULL\": invalid identifier")) {
+                if (e.getMessage().contains("ORA-00904")) {
```

Comment:
Can't we use `SQLException#getSQLState()`? Is it set properly by the ODBC driver? It should contain exactly and only this error code.

Reply:
The problem with SQL State is that we get `42000` which implies somewhat of a myriad of reasons as to why the SQL statement failed, including but not limited to insufficient permissions from what I have concluded.  If there is a situation where Oracle throws an insufficient permission problem, we should explicitly fail.  The only time we want to allow the code to move forward is explicitly if that field isn't detected.